test/torchaudio_unittest/prototype/hdemucs_test_impl.py

```diff
@@ -106,3 +116,88 @@ def test_decoder_output_shape_time(self):
 
         assert z.size() == (batch_size, chout, t * stride)
         assert y.size() == (batch_size, chin, t)
+
+
+@skipIfNoModule("demucs")
+class CompareHDemucsOriginal(TorchaudioTestCase):
+    """Test the process of importing the models from demucs.
+
+    Test methods in this test suite will check to
+    1. Assure correctness in factory functions, comparing with original hybrid demucs
+    """
+
+    def _get_original_model(self, sources: List[str], nfft: int, depth: int):
+        from demucs import hdemucs as original
+
+        original = original.HDemucs(sources, nfft=nfft, depth=depth)
+        return original
+
+    @parameterized.expand(
+        [
+            (["bass", "drums", "other", "vocals"],),
+            (["bass", "drums", "other"],),
+            (["bass", "vocals"],),
+            (["vocals"],),
+        ]
+    )
+    def test_import_recreate_low_model_test(self, sources):
+        sample_rate = 8000
+        nfft = 1024
+        depth = 5
+        duration = 10
+        channels = 2
+        batch_size = 1
+
+        torch.random.manual_seed(0)
+        factory_hdemucs = hdemucs_low(sources, sample_rate=sample_rate).eval()
+        self._assert_equal_models(batch_size, channels, depth, duration, factory_hdemucs, nfft, sample_rate, sources)
+
+    @parameterized.expand(
+        [
+            (["bass", "drums", "other", "vocals"],),
+            (["bass", "drums", "other"],),
+            (["bass", "vocals"],),
+            (["vocals"],),
+        ]
+    )
+    def test_import_recreate_medium_model_test(self, sources):
+        sample_rate = 16000
+        nfft = 2048
+        depth = 6
+        duration = 10
+        channels = 2
+        batch_size = 1
+
+        torch.random.manual_seed(0)
+        factory_hdemucs = hdemucs_medium(sources, sample_rate=sample_rate).eval()
+        self._assert_equal_models(batch_size, channels, depth, duration, factory_hdemucs, nfft, sample_rate, sources)
+
+    @parameterized.expand(
+        [
+            (["bass", "drums", "other", "vocals"],),
+            (["bass", "drums", "other"],),
+            (["bass", "vocals"],),
+            (["vocals"],),
+        ]
+    )
+    def test_import_recreate_high_model_test(self, sources):
+        sample_rate = 44100
+        nfft = 4096
+        depth = 6
+        duration = 10
+        channels = 2
+        batch_size = 1
+
+        torch.random.manual_seed(0)
+        factory_hdemucs = hdemucs_high(sources, sample_rate=sample_rate).eval()
+        self._assert_equal_models(batch_size, channels, depth, duration, factory_hdemucs, nfft, sample_rate, sources)
+
+    def _assert_equal_models(self, batch_size, channels, depth, duration, factory_hdemucs, nfft, sample_rate, sources):
```

Comment:
can you move helper functions to the top of the class, and move `factory_hdemucs` to the first arg rather than have it in the middle of model params